core/src/main/java/org/elasticsearch/index/percolator/PercolatorQueriesRegistry.java

```diff
@@ -208,12 +208,12 @@ private Query parseQuery(String type, XContentParser parser) {
             context.setMapUnmappedFieldAsString(mapUnmappedFieldsAsString ? true : false);
             return queryParserService.parseInnerQuery(context);
         } catch (IOException e) {
-            throw new QueryParsingException(context, "Failed to parse", e);
+            throw new QueryCreationException(context, "Failed to parse", e);
```

Comment:
Haven't realized this so far, this also means that we will not be able to remove the parser part from whatever context object we use here since we need both functionalitites still. Any feeling about how this might eventually change after the refactoring? Will we pass builders to this method then?